### ISSUE TITLE
tf.keras.losses.categorical_crossentropy ignores batch size and dont have name keywordarg

### ISSUE THREAD

**Issue by Uiuran** [tensorflow/tensorflow#30780]
I think this function is obsolete in Tensorflow since the class CategoricalCrossEntropy does the job in the standard way (i.e, compute loss by batch and associate with a name to be easier to debug). 

I suggest to drop this from next 1.14 version if iam not taking something wrong. 

Thank you.

**Comment by pavithrasv** [comment]
The class version compute the per-sample loss values and then applies sample weight, reduces loss as per the reduction specified. The categorical_crossentropy function just computes the per-sample loss values. We would like to keep all the class as well as the function versions of losses.
